### PR TITLE
Switch from Minimist to SHON

### PR DESCRIPTION
**Pull request by kriskowal**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -148,14 +127,10 @@ main.exec = function execMain(str, delegate) {
 };
 
 function help() {
-    console.log('usage: tcurl [--help] [-v | --version] [-H] [-p] [-t]');
-    console.log('             [-2 | --arg2 | --head] [-3 | --arg3 | --body]');
-    console.log('             [--shardKey] [--no-strict]  [--timeout]');
-    console.log('             [--http] [--raw] [--health]');
-    console.log('             [--rate] [--requests] [--time] [--delay]');
+    logUsage(command);
```
Comment (Raynos) [inline index.js]: Can shon output short help ?

**Comment by Raynos** [comment]
We should probably update manpage too.

File: index.js (hunk 1 above)
Comment (kriskowal) [inline index.js]: It does not yet have an abbreviator, but it could.

**Comment by kriskowal** [comment]
I’ve upgraded this from WIP to the real deal™. Updated the man page (and the formatting errors in the man page). I’ve decided to dump the usage on `-h` since I doubt the flag summary is useful without any description. The flag summary is retained in the man page.

This is messy, but to make migration smooth, you can pass JSON with the flagged argument like `--body {"nuck":{"language":"js","name":"tcurl"}}` or use the positional argument for SHON like `[ --nyuck [ --language js --name tcurl ] ]`.

Backward incompatibility is an error. Please note any you find.

Please review by running the branch anywhere you are using tcurl. Please note the errors for invalid usage. Please inform me of ergonomic issues.

In a future change, I intend to introduce `-d raw-data` or `-d @file` or `-d-` notation for as=raw, which will support streaming.

r @malandrew @Raynos @jcorbin @blampe @abhinav @breerly @ShanniLi

**Comment by kriskowal** [comment]
Verified health check in production with this branch.

**Pull request merged**